File: server/src/domain/glossary/glossaryController.js
// In-Memory store for Glossary CRUD (Backend B - F-5)
let glossaries = [
  { id: '1', term: 'Manyfast', rule: '원문 유지 (Keep Original)', note: '의역 금지' },
  { id: '2', term: 'ASAP', rule: '오늘 EOD 18:00 전', note: '팀 내 합의 기준' }
];

export function getGlossaries(req, res) {
  res.json({ data: glossaries });
}

export function createGlossary(req, res) {
  const { term, rule, note } = req.body;
  const newEntry = { id: String(Date.now()), term, rule: rule || '원문 유지', note: note || '' };
  glossaries.push(newEntry);
  res.status(201).json({ data: newEntry });
}

export function deleteGlossary(req, res) {
  const { id } = req.params;
  glossaries = glossaries.filter(g => g.id !== id);
  res.json({ message: 'Glossary entry deleted successfully' });
}

File: server/src/main/java/com/likelion/manyfast/domain/glossary/GlossaryController.java
package com.likelion.manyfast.domain.glossary;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import java.util.*;

@RestController
@RequestMapping("/api/glossaries")
public class GlossaryController {

    private final List<Map<String, Object>> glossaries = new ArrayList<>(List.of(
            Map.of("id", "1", "term", "Manyfast", "rule", "원문 유지 (Keep Original)", "note", "의역 금지"),
            Map.of("id", "2", "term", "ASAP", "rule", "오늘 EOD 18:00 전", "note", "팀 내 합의 기준")
    ));

    @GetMapping
    public ResponseEntity<Map<String, Object>> getGlossaries() {
        return ResponseEntity.ok(Map.of("data", glossaries));
    }

    @PostMapping
    public ResponseEntity<Map<String, Object>> createGlossary(@RequestBody Map<String, Object> request) {
        Map<String, Object> newEntry = new HashMap<>(request);
        newEntry.put("id", String.valueOf(System.currentTimeMillis()));
        glossaries.add(newEntry);
        return ResponseEntity.status(201).body(Map.of("data", newEntry));
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Map<String, String>> deleteGlossary(@PathVariable String id) {
        glossaries.removeIf(g -> Objects.equals(g.get("id"), id));
        return ResponseEntity.ok(Map.of("message", "Glossary entry deleted successfully"));
    }
}
